<a href="https://colab.research.google.com/github/higginbotham-thomas/Course2/blob/main/Credit%20One%20Initial%20Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
#for colab, install pymysql and pandas_profiling
#!pip3 install pymysql
#!pip3 uninstall pandas-profiling
#!pip3 install pandas-profiling

In [1]:
#imports
from sqlalchemy import create_engine
import pymysql
import pandas as pd
import numpy as np
from pandas_profiling import ProfileReport

In [2]:
#define database connection
connection = pymysql.connect(host='data-analytics-2018.cbrosir2cswx.us-east-1.rds.amazonaws.com',
user='deepAnalytics',
password='Sqltask1234!',
database='Credit',
charset='utf8mb4',
cursorclass=pymysql.cursors.DictCursor)

In [3]:
#dataframe
df = pd.read_sql('SELECT * FROM credit', con=connection)

C:\Users\higgi\.conda\envs\Course2\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


In [5]:
df.iloc[0]

X1                      LIMIT_BAL
X2                            SEX
X3                      EDUCATION
X4                       MARRIAGE
X5                            AGE
X6                          PAY_0
X7                          PAY_2
X8                          PAY_3
X9                          PAY_4
X10                         PAY_5
X11                         PAY_6
X12                     BILL_AMT1
X13                     BILL_AMT2
X14                     BILL_AMT3
X15                     BILL_AMT4
X16                     BILL_AMT5
X17                     BILL_AMT6
X18                      PAY_AMT1
X19                      PAY_AMT2
X20                      PAY_AMT3
X21                      PAY_AMT4
X22                      PAY_AMT5
X23                      PAY_AMT6
Y      default payment next month
Name: 0, dtype: object

In [5]:
#make the first row into column headers
df = df.rename(columns=df.iloc[0]).drop(df.index[0])

In [6]:
#rename PAY_0 column to PAY_1
df.rename(columns = {'PAY_0':'PAY_1'}, inplace = True)

In [7]:
#remove duplicate headers
df = df.loc[df['LIMIT_BAL'] != 'LIMIT_BAL']

In [8]:
#check the first 5 rows of the data frame
df.head()

LIMIT_BAL     SEX   EDUCATION MARRIAGE AGE PAY_1 PAY_2 PAY_3 PAY_4 PAY_5  \
1     20000  female  university        1  24     2     2    -1    -1    -2   
2    120000  female  university        2  26    -1     2     0     0     0   
3     90000  female  university        2  34     0     0     0     0     0   
4     50000  female  university        1  37     0     0     0     0     0   
5     50000    male  university        1  57    -1     0    -1     0     0   

   ... BILL_AMT4 BILL_AMT5 BILL_AMT6 PAY_AMT1 PAY_AMT2 PAY_AMT3 PAY_AMT4  \
1  ...         0         0         0        0      689        0        0   
2  ...      3272      3455      3261        0     1000     1000     1000   
3  ...     14331     14948     15549     1518     1500     1000     1000   
4  ...     28314     28959     29547     2000     2019     1200     1100   
5  ...     20940     19146     19131     2000    36681    10000     9000   

  PAY_AMT5 PAY_AMT6 default payment next month  
1        0        0                    default  
2        0     2000                    default  
3     1000     5000                not default  
4     1069     1000                not default  
5      689      679                not default  

[5 rows x 24 columns]

In [9]:
#show the number of values and data types for each column
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3668 entries, 1 to 3669
Data columns (total 24 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   LIMIT_BAL                   3668 non-null   object
 1   SEX                         3668 non-null   object
 2   EDUCATION                   3668 non-null   object
 3   MARRIAGE                    3668 non-null   object
 4   AGE                         3668 non-null   object
 5   PAY_1                       3668 non-null   object
 6   PAY_2                       3668 non-null   object
 7   PAY_3                       3668 non-null   object
 8   PAY_4                       3668 non-null   object
 9   PAY_5                       3668 non-null   object
 10  PAY_6                       3668 non-null   object
 11  BILL_AMT1                   3668 non-null   object
 12  BILL_AMT2                   3668 non-null   object
 13  BILL_AMT3                   3668 non-null   obje

In [10]:
#check the column names
df.columns

Index(['LIMIT_BAL', 'SEX', 'EDUCATION', 'MARRIAGE', 'AGE', 'PAY_1', 'PAY_2',
       'PAY_3', 'PAY_4', 'PAY_5', 'PAY_6', 'BILL_AMT1', 'BILL_AMT2',
       'BILL_AMT3', 'BILL_AMT4', 'BILL_AMT5', 'BILL_AMT6', 'PAY_AMT1',
       'PAY_AMT2', 'PAY_AMT3', 'PAY_AMT4', 'PAY_AMT5', 'PAY_AMT6',
       'default payment next month'],
      dtype='object')

In [11]:
#basic statistics for each column of the data
df.describe()

LIMIT_BAL     SEX   EDUCATION MARRIAGE   AGE PAY_1 PAY_2 PAY_3 PAY_4  \
count       3668    3668        3668     3668  3668  3668  3668  3668  3668   
unique        62       2           4        4    52     9    10    10     9   
top        50000  female  university        2    29     0     0     0     0   
freq         453    2130        1644     2045   214  1741  1901  1875  1995   

       PAY_5  ... BILL_AMT4 BILL_AMT5 BILL_AMT6 PAY_AMT1 PAY_AMT2 PAY_AMT3  \
count   3668  ...      3668      3668      3668     3668     3668     3668   
unique     8  ...      2008      1983      1947     1146     1129     1040   
top        0  ...         0         0         0        0        0        0   
freq    1996  ...       424       460       532      667      708      798   

       PAY_AMT4 PAY_AMT5 PAY_AMT6 default payment next month  
count      3668     3668     3668                       3668  
unique     1034     1038      970                          2  
top           0        0        0                not default  
freq        808      827      949                       2873  

[4 rows x 24 columns]

In [11]:
#basic statistics for each column of the data
df.describe()

LIMIT_BAL     SEX   EDUCATION MARRIAGE   AGE PAY_1 PAY_2 PAY_3 PAY_4  \
count       3668    3668        3668     3668  3668  3668  3668  3668  3668   
unique        62       2           4        4    52     9    10    10     9   
top        50000  female  university        2    29     0     0     0     0   
freq         453    2130        1644     2045   214  1741  1901  1875  1995   

       PAY_5  ... BILL_AMT4 BILL_AMT5 BILL_AMT6 PAY_AMT1 PAY_AMT2 PAY_AMT3  \
count   3668  ...      3668      3668      3668     3668     3668     3668   
unique     8  ...      2008      1983      1947     1146     1129     1040   
top        0  ...         0         0         0        0        0        0   
freq    1996  ...       424       460       532      667      708      798   

       PAY_AMT4 PAY_AMT5 PAY_AMT6 default payment next month  
count      3668     3668     3668                       3668  
unique     1034     1038      970                          2  
top           0        0        0                not default  
freq        808      827      949                       2873  

[4 rows x 24 columns]

In [12]:
#drop any duplicate records
df = df.drop_duplicates()

In [13]:
#replace spaces in column names
df.columns = df.columns.str.replace(' ', '_')

In [14]:
#factorize sex column
df.loc[df['SEX'] == 'male', 'SEX'] = 0
df.loc[df['SEX'] == 'female', 'SEX'] = 1

#factorize education column
df.loc[df['EDUCATION'] == 'high school', 'EDUCATION'] = 0
df.loc[df['EDUCATION'] == 'university', 'EDUCATION'] = 1
df.loc[df['EDUCATION'] == 'graduate school', 'EDUCATION'] = 2
df.loc[df['EDUCATION'] == 'other', 'EDUCATION'] = 3

#factorize default_payment_next_month column
df.loc[df['default_payment_next_month'] == 'default', 'default_payment_next_month'] = 0
df.loc[df['default_payment_next_month'] == 'not default', 'default_payment_next_month'] = 1

In [15]:
#unique values for a column
df.default_payment_next_month.unique()

array([0, 1], dtype=object)

In [16]:
#convert all values in dataframe to int
df = df.astype (int)

In [17]:
#find the number of months a payment was reported defaulted
df['DEFAULTED_NMOS']=df['PAY_1'].apply(lambda x: 1 if x >= 1 else 0)+df['PAY_2'].apply(lambda x: 1 if x >= 1 else 0)+df['PAY_3'].apply(lambda x: 1 if x >= 1 else 0)+df['PAY_4'].apply(lambda x: 1 if x >= 1 else 0)+df['PAY_5'].apply(lambda x: 1 if x >= 1 else 0)+df['PAY_6'].apply(lambda x: 1 if x >= 1 else 0)


In [18]:
#find whether a default occurred AT ALL during the reporting periods
df['DEFAULTED'] = np.where(df.DEFAULTED_NMOS > 0, 1, 0)

In [19]:
#basic statistics for each column of the data (following drop duplicates)
df.describe()

LIMIT_BAL          SEX    EDUCATION     MARRIAGE          AGE  \
count     2396.000000  2396.000000  2396.000000  2396.000000  2396.000000   
mean    165834.724541     0.577629     1.230801     1.580134    35.419866   
std     129493.683277     0.494040     0.721101     0.525593     9.452344   
min      10000.000000     0.000000     0.000000     0.000000    21.000000   
25%      50000.000000     0.000000     1.000000     1.000000    28.000000   
50%     140000.000000     1.000000     1.000000     2.000000    34.000000   
75%     230000.000000     1.000000     2.000000     2.000000    41.000000   
max    1000000.000000     1.000000     3.000000     3.000000    75.000000   

             PAY_1        PAY_2        PAY_3        PAY_4        PAY_5  ...  \
count  2396.000000  2396.000000  2396.000000  2396.000000  2396.000000  ...   
mean      0.004591    -0.136477    -0.151920    -0.249165    -0.259599  ...   
std       1.123473     1.207902     1.247411     1.176359     1.156723  ...   
min      -2.000000    -2.000000    -2.000000    -2.000000    -2.000000  ...   
25%      -1.000000    -1.000000    -1.000000    -1.000000    -1.000000  ...   
50%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
75%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
max       8.000000     7.000000     7.000000     7.000000     7.000000  ...   

           BILL_AMT6       PAY_AMT1       PAY_AMT2       PAY_AMT3  \
count    2396.000000    2396.000000    2396.000000    2396.000000   
mean    38797.298414    5548.942821    5220.050501    4855.443656   
std     65468.199929   13701.948666   14616.927369   22156.285391   
min   -339603.000000       0.000000       0.000000       0.000000   
25%      1012.000000     997.500000     588.000000     247.750000   
50%     15819.500000    2117.500000    2000.000000    1443.000000   
75%     47369.500000    5002.250000    4921.750000    4000.000000   
max    961664.000000  239104.000000  285138.000000  896040.000000   

            PAY_AMT4       PAY_AMT5       PAY_AMT6  \
count    2396.000000    2396.000000    2396.000000   
mean     4891.429883    4984.977880    5149.928631   
std     14252.068389   16896.223773   20076.659470   
min         0.000000       0.000000       0.000000   
25%       240.000000     247.750000       0.000000   
50%      1500.000000    1500.000000    1328.000000   
75%      4000.000000    4000.000000    4000.000000   
max    205000.000000  332000.000000  528666.000000   

       default_payment_next_month  DEFAULTED_NMOS    DEFAULTED  
count                 2396.000000     2396.000000  2396.000000  
mean                     0.783389        0.846828     0.347663  
std                      0.412021        1.563518     0.476328  
min                      0.000000        0.000000     0.000000  
25%                      1.000000        0.000000     0.000000  
50%                      1.000000        0.000000     0.000000  
75%                      1.000000        1.000000     1.000000  
max                      1.000000        6.000000     1.000000  

[8 rows x 26 columns]

There were 3670 values before dropping duplicates. Afterward, there were 2396 values.

In [20]:
df.cov()

LIMIT_BAL          SEX     EDUCATION  \
LIMIT_BAL                   1.676861e+10   837.052708  21466.971745   
SEX                         8.370527e+02     0.244076     -0.016881   
EDUCATION                   2.146697e+04    -0.016881      0.519986   
MARRIAGE                   -1.005857e+04    -0.007476      0.056028   
AGE                         1.896911e+05    -0.547430     -1.351226   
PAY_1                      -2.917502e+04    -0.019772     -0.101686   
PAY_2                      -3.882758e+04    -0.038044     -0.143853   
PAY_3                      -3.815289e+04    -0.031208     -0.136112   
PAY_4                      -3.417397e+04    -0.022194     -0.116999   
PAY_5                      -3.197111e+04    -0.025351     -0.101228   
PAY_6                      -3.227379e+04    -0.023600     -0.108550   
BILL_AMT1                   3.648630e+09 -2164.854740    261.268977   
BILL_AMT2                   3.532902e+09 -1663.790584    676.372766   
BILL_AMT3                   3.297613e+09 -1476.905788    401.747393   
BILL_AMT4                   3.313588e+09 -1462.298256    900.985570   
BILL_AMT5                   3.046906e+09 -1235.106043    575.589439   
BILL_AMT6                   3.022809e+09 -1375.511484    292.248425   
PAY_AMT1                    4.684651e+08    99.100683    774.496710   
PAY_AMT2                    5.096867e+08    14.368729    621.260156   
PAY_AMT3                    6.704545e+08    30.170764    790.334722   
PAY_AMT4                    3.584094e+08   -64.481402    611.629760   
PAY_AMT5                    4.871168e+08  -301.644837    735.809700   
PAY_AMT6                    5.412305e+08  -205.532036    845.703327   
default_payment_next_month  4.500301e+03     0.001165      0.017865   
DEFAULTED_NMOS             -4.014140e+04    -0.023386     -0.096574   
DEFAULTED                  -1.071412e+04    -0.000487     -0.015974   

                                MARRIAGE            AGE         PAY_1  \
LIMIT_BAL                  -10058.566644  189691.134493 -29175.023787   
SEX                            -0.007476      -0.547430     -0.019772   
EDUCATION                       0.056028      -1.351226     -0.101686   
MARRIAGE                        0.276248      -2.087939      0.015290   
AGE                            -2.087939      89.346812     -0.189402   
PAY_1                           0.015290      -0.189402      1.262192   
PAY_2                           0.023676      -0.341421      0.865345   
PAY_3                           0.025540      -0.445582      0.767712   
PAY_4                           0.028952      -0.273211      0.657512   
PAY_5                           0.027074      -0.138139      0.614136   
PAY_6                           0.025541      -0.105708      0.591692   
BILL_AMT1                   -1655.601465   50246.908772  15016.264546   
BILL_AMT2                   -1494.415030   43956.155322  14863.644606   
BILL_AMT3                   -1731.488883   43395.642119  13569.784507   
BILL_AMT4                   -1365.926518   37803.951152  13319.744428   
BILL_AMT5                   -1318.268027   31681.465104  13072.301012   
BILL_AMT6                   -1489.604090   39632.598662  13002.822429   
PAY_AMT1                     -280.664936    4642.480802  -1604.735437   
PAY_AMT2                     -520.449351    9867.140374  -1922.451171   
PAY_AMT3                     -308.626590    8479.786089   -773.618322   
PAY_AMT4                      -76.286655    -979.759275   -986.931828   
PAY_AMT5                     -327.827663    4248.225575   -798.476308   
PAY_AMT6                     -193.856700    7159.267139  -1489.141217   
default_payment_next_month      0.005465      -0.154526     -0.133452   
DEFAULTED_NMOS                  0.012071      -0.000797      1.097154   
DEFAULTED                       0.004070      -0.111795      0.325752   

                                   PAY_2         PAY_3         PAY_4  \
LIMIT_BAL                  -38827.576232 -38152.885289 -34173.974718 

In [21]:
df.corr()

LIMIT_BAL       SEX  EDUCATION  MARRIAGE  \
LIMIT_BAL                    1.000000  0.013084   0.229893 -0.147788   
SEX                          0.013084  1.000000  -0.047384 -0.028791   
EDUCATION                    0.229893 -0.047384   1.000000  0.147828   
MARRIAGE                    -0.147788 -0.028791   0.147828  1.000000   
AGE                          0.154974 -0.117227  -0.198241 -0.420270   
PAY_1                       -0.200540 -0.035623  -0.125517  0.025893   
PAY_2                       -0.248233 -0.063752  -0.165155  0.037293   
PAY_3                       -0.236194 -0.050640  -0.151318  0.038955   
PAY_4                       -0.224340 -0.038189  -0.137926  0.046826   
PAY_5                       -0.213442 -0.044360  -0.121360  0.044533   
PAY_6                       -0.210881 -0.040419  -0.127372  0.041117   
BILL_AMT1                    0.361823 -0.056271   0.004653 -0.040450   
BILL_AMT2                    0.361342 -0.044604   0.012423 -0.037658   
BILL_AMT3                    0.365711 -0.042932   0.008001 -0.047310   
BILL_AMT4                    0.375659 -0.043453   0.018343 -0.038152   
BILL_AMT5                    0.365777 -0.038864   0.012409 -0.038991   
BILL_AMT6                    0.356559 -0.042528   0.006191 -0.043290   
PAY_AMT1                     0.264026  0.014640   0.078387 -0.038972   
PAY_AMT2                     0.269277  0.001990   0.058942 -0.067744   
PAY_AMT3                     0.233681  0.002756   0.049467 -0.026502   
PAY_AMT4                     0.194202 -0.009158   0.059513 -0.010184   
PAY_AMT5                     0.222636 -0.036136   0.060392 -0.036915   
PAY_AMT6                     0.208182 -0.020722   0.058416 -0.018371   
default_payment_next_month   0.084348  0.005722   0.060128  0.025237   
DEFAULTED_NMOS              -0.198263 -0.030276  -0.085657  0.014688   
DEFAULTED                   -0.173701 -0.002068  -0.046506  0.016259   

                                 AGE     PAY_1     PAY_2     PAY_3     PAY_4  \
LIMIT_BAL                   0.154974 -0.200540 -0.248233 -0.236194 -0.224340   
SEX                        -0.117227 -0.035623 -0.063752 -0.050640 -0.038189   
EDUCATION                  -0.198241 -0.125517 -0.165155 -0.151318 -0.137926   
MARRIAGE                   -0.420270  0.025893  0.037293  0.038955  0.046826   
AGE                         1.000000 -0.017835 -0.029903 -0.037790 -0.024571   
PAY_1                      -0.017835  1.000000  0.637668  0.547805  0.497509   
PAY_2                      -0.029903  0.637668  1.000000  0.763806  0.667189   
PAY_3                      -0.037790  0.547805  0.763806  1.000000  0.812452   
PAY_4                      -0.024571  0.497509  0.667189  0.812452  1.000000   
PAY_5                      -0.012634  0.472577  0.639244  0.724154  0.835864   
PAY_6                      -0.009462  0.445626  0.606825  0.669614  0.742573   
BILL_AMT1                   0.068263  0.171638  0.225230  0.195056  0.200938   
BILL_AMT2                   0.061591  0.175226  0.229758  0.220140  0.221052   
BILL_AMT3                   0.065932  0.173459  0.232926  0.224046  0.253484   
BILL_AMT4                   0.058714  0.174051  0.229723  0.221919  0.249450   
BILL_AMT5                   0.052104  0.180882  0.233963  0.223246  0.254886   
BILL_AMT6                   0.064045  0.176785  0.223571  0.217238  0.244399   
PAY_AMT1                    0.035845 -0.104246 -0.067889 -0.003731 -0.014056   
PAY_AMT2                    0.071416 -0.117068 -0.069403 -0.091800  0.007803   
PAY_AMT3                    0.040490 -0.031079 -0.016359 -0.018293 -0.044739   
PAY_AMT4                   -0.007273 -0.061638 -0.041619 -0.049315 -0.026550   
PAY_AMT5                    0.026600 -0.042064 -0.036875 -0.032942 -0.022711   
PAY_AMT6                    0.037726 -0.066021 -0.025573 -0.027555 -0.009108   
default_payment_next_month -0.039677 -0.288299 -0.202049 -0.206223 -0.172565   
DEFAULTED_NMOS             -0.000054  0.624601  0.651298  0.665637  0.632132   
DEF

In [22]:
df.describe()

LIMIT_BAL          SEX    EDUCATION     MARRIAGE          AGE  \
count     2396.000000  2396.000000  2396.000000  2396.000000  2396.000000   
mean    165834.724541     0.577629     1.230801     1.580134    35.419866   
std     129493.683277     0.494040     0.721101     0.525593     9.452344   
min      10000.000000     0.000000     0.000000     0.000000    21.000000   
25%      50000.000000     0.000000     1.000000     1.000000    28.000000   
50%     140000.000000     1.000000     1.000000     2.000000    34.000000   
75%     230000.000000     1.000000     2.000000     2.000000    41.000000   
max    1000000.000000     1.000000     3.000000     3.000000    75.000000   

             PAY_1        PAY_2        PAY_3        PAY_4        PAY_5  ...  \
count  2396.000000  2396.000000  2396.000000  2396.000000  2396.000000  ...   
mean      0.004591    -0.136477    -0.151920    -0.249165    -0.259599  ...   
std       1.123473     1.207902     1.247411     1.176359     1.156723  ...   
min      -2.000000    -2.000000    -2.000000    -2.000000    -2.000000  ...   
25%      -1.000000    -1.000000    -1.000000    -1.000000    -1.000000  ...   
50%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
75%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
max       8.000000     7.000000     7.000000     7.000000     7.000000  ...   

           BILL_AMT6       PAY_AMT1       PAY_AMT2       PAY_AMT3  \
count    2396.000000    2396.000000    2396.000000    2396.000000   
mean    38797.298414    5548.942821    5220.050501    4855.443656   
std     65468.199929   13701.948666   14616.927369   22156.285391   
min   -339603.000000       0.000000       0.000000       0.000000   
25%      1012.000000     997.500000     588.000000     247.750000   
50%     15819.500000    2117.500000    2000.000000    1443.000000   
75%     47369.500000    5002.250000    4921.750000    4000.000000   
max    961664.000000  239104.000000  285138.000000  896040.000000   

            PAY_AMT4       PAY_AMT5       PAY_AMT6  \
count    2396.000000    2396.000000    2396.000000   
mean     4891.429883    4984.977880    5149.928631   
std     14252.068389   16896.223773   20076.659470   
min         0.000000       0.000000       0.000000   
25%       240.000000     247.750000       0.000000   
50%      1500.000000    1500.000000    1328.000000   
75%      4000.000000    4000.000000    4000.000000   
max    205000.000000  332000.000000  528666.000000   

       default_payment_next_month  DEFAULTED_NMOS    DEFAULTED  
count                 2396.000000     2396.000000  2396.000000  
mean                     0.783389        0.846828     0.347663  
std                      0.412021        1.563518     0.476328  
min                      0.000000        0.000000     0.000000  
25%                      1.000000        0.000000     0.000000  
50%                      1.000000        0.000000     0.000000  
75%                      1.000000        1.000000     1.000000  
max                      1.000000        6.000000     1.000000  

[8 rows x 26 columns]

In [23]:
#profile report
sample = df.sample(frac=1.0)
profile = ProfileReport(sample)

In [24]:
profile.to_file("Credit_One_Pandas_Profile_Report.html")

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

In [25]:
df.to_csv("Credit_One_Data.csv")

In [26]:
df

LIMIT_BAL  SEX  EDUCATION  MARRIAGE  AGE  PAY_1  PAY_2  PAY_3  PAY_4  \
1         20000    1          1         1   24      2      2     -1     -1   
2        120000    1          1         2   26     -1      2      0      0   
3         90000    1          1         2   34      0      0      0      0   
4         50000    1          1         1   37      0      0      0      0   
5         50000    0          1         1   57     -1      0     -1      0   
...         ...  ...        ...       ...  ...    ...    ...    ...    ...   
2393      50000    0          0         1   32      2      3      2      2   
2394      20000    1          0         2   49      0      0      2      0   
2395     130000    1          1         2   24      1     -2     -1     -1   
2396     110000    1          0         1   27      0      0      0      0   
2397     200000    0          1         1   29      0      0      0      2   

      PAY_5  ...  BILL_AMT6  PAY_AMT1  PAY_AMT2  PAY_AMT3  PAY_AMT4  PAY_AMT5  \
1        -2  ...          0         0       689         0         0         0   
2         0  ...       3261         0      1000      1000      1000         0   
3         0  ...      15549      1518      1500      1000      1000      1000   
4         0  ...      29547      2000      2019      1200      1100      1069   
5         0  ...      19131      2000     36681     10000      9000       689   
...     ...  ...        ...       ...       ...       ...       ...       ...   
2393      2  ...      44960      1500      1600      1700      1500      3600   
2394     -1  ...          0      3008         0       600      1150         0   
2395     -1  ...        937         0      3372         0      1001       939   
2396      0  ...      47141      2326      3000      1807      1698      1911   
2397      2  ...      47539      2078      3900      1900      1904        12   

      PAY_AMT6  default_payment_next_month  DEFAULTED_NMOS  DEFAULTED  
1            0                           0               2          1  
2         2000                           0               2          1  
3         5000                           1               0          0  
4         1000                           1               0          0  
5          679                           1               0          0  
...        ...                         ...             ...        ...  
2393         0                           0               6          1  
2394         0                           1               1          1  
2395      1013                           1               1          1  
2396      1808                           1               0          0  
2397      5224                           1               3          1  

[2396 rows x 26 columns]